### DISCUSSION TITLE
walrus S3

### DISCUSSION THREAD

**Discussion by 0xWrapper** [MystenLabs/walrus-docs#87] [Ideas]
I came up with an interesting concept for implementing the S3 protocol on Walrus

1. A deployed S3 endpoint is built using nextjs, and the Walrus proxy is S3
2. An easy-to-use object system usage platform using zkLogin

**Comment by Euraxluo** [comment]
Can Walrus be given a subcommand to provide an s3 gateway and support accessing endpoints using Sui signatures?

Given that hackathon already has an s3 cli, Walrus' native S3 endpoint will be a new product form
